R/estimatr_lh_robust.R

```diff
@@ -0,0 +1,114 @@
+#' Linear Hypothesis for Ordinary Least Squares with Robust Standard Errors
+#'
+#' @description This function fits a linear model with robust standard errors and performs linear hypothesis test.
+#' @param ... Other arguments to be passed to  \code{\link{lm_robust}}
+#' @param data A \code{data.frame}
+#' @param linear_hypothesis A character string or a matrix specifying combination, to be passed to the hypothesis.matrix argument of car::linearHypothesis
+#' See \code{\link[car]{linearHypothesis}} for more details.
+#' @details
+#'
+#' This function is a wrapper for \code{\link{lm_robust}} and for
+#' \code{\link[car]{linearHypothesis}}. It first runs \code{lm_robust} and
+#' next passes \code{"lm_robust"} object as an argument to \code{linearHypothesis}.
+#'
+#' @return An object of class \code{"lh_robust"} containing the two following components:
+#'
+#' \item{lm_robust}{an object as returned by \code{lm_robust}.}
+#' \item{lh}{A data frame with most of its columns pulled from \code{linearHypothesis}' output.}
+#'
+#' The only analyis directly performed by \code{lh_robust} is a \code{t-test} for the null hypothesis of no effects of the linear combination of coefficients as specified by the user.
+#' All other output components are either extract from \code{linearHypothesis} or \code{lm_robust}.
+#'
+#' The original output returned by \code{linearHypothesis} is added as an attribute.
+#'
+#' @examples
+#'
+#' library(fabricatr)
+#' dat <- fabricate(
+#'   N = 40,
+#'   y = rpois(N, lambda = 4),
+#'   x = rnorm(N),
+#'   z = rbinom(N, 1, prob = 0.4),
+#'   clusterID = sample(1:4, 40, replace = TRUE)
+#' )
+#'
+#' # Default variance estimator is HC2 robust standard errors
+#' lhro <- lh_robust(y ~ x + z, data = dat, linear_hypothesis = "z + 2x = 0")
+#'
+#' # The linear hypothesis argument can be specified as following
+#' lh_robust(y ~ x + z, data = dat, linear_hypothesis = "z = 2x")
+#' lh_robust(y ~ x + z, data = dat, linear_hypothesis = c("z = 1", "x = 2"))
+#' lh_robust(y ~ x + z, data = dat, linear_hypothesis = "2*x +1*z")
+#'
+#' # which are alle equivalent to
```

Comment:
Typo, consider just having them in one block and say they are all equivalent.

Reply:
thx